modules/service/src/main/scala/lucuma/odb/graphql/mapping/StepRecordMapping.scala

```diff
@@ -54,6 +54,7 @@ trait StepRecordMapping[F[_]] extends StepRecordView[F]
         SqlField("instrument",   StepRecordView.Instrument, discriminator = true),
         SqlObject("atom",        Join(StepRecordView.AtomId, AtomRecordTable.Id)),
         SqlField("created",      StepRecordView.Created),
+        EffectField("executionState", executionStateHandler, List("id")),
```

Comment:
This is accomplished via a query in an `EffectHandler`.  Another alternative would be to update the step record as events are received, storing the updated execution state in the `t_step_record` table as we go and avoiding a more costly query.  That requires a couple more moving parts though.

Reply:
:+1: Seems reasonable to do it the "easy" way and add more caching if it's a problem.